### PR TITLE
Audit3. Convert random buckets selection to user defined buckets.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
We remove a random selection of the storage and distribution buckets on dynamic bags creation from the storage pallet and replace it with user-defined buckets.

#### Changes
- removed a random selection of the buckets
- added new parameters (buckets) to the bag creation methods
- modified extrinsics for dynamic bag creation policy modifications (enforced new checks for parameters)
- fixed tests for storage and content pallets.

[Original issue](https://github.com/Joystream/joystream/issues/3289)

### CODE REVIEW

**Comment by shamil-gadelshin** [comment]
We need to update the `audit3` branch with the latest `olympia` code in order to review this PR properly.

File: runtime-modules/storage/src/lib.rs [hunk 1]
```diff
@@ -2978,6 +2958,25 @@ impl<T: Trait> Module<T> {
             Self::check_buckets_for_overflow(storage_buckets, &bag_change.voucher_update)?;
         }
 
+        // Validate distribution bucket IDs
```
Comment (dobertRowneySr) [inline runtime-modules/storage/src/lib.rs]: This has become a long function now with several sub validation routines. Can you please refactor each unit with self contained validation logic (like distribution bucket validation) into its own separate function. I think it is clearer that way

**Review by dobertRowneySr — CHANGES_REQUESTED** [review]
It's ok. No major issue found. I just left a small request in order to enhance code clarity

File: runtime-modules/storage/src/lib.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/storage/src/lib.rs]: Fixed here: https://github.com/Joystream/joystream/pull/3348/commits/32d252427ebd6c3b417662badd523e572ab3cbc4

**Review by dobertRowneySr — APPROVED** [review]
LGTM

**Pull request merged**